homeassistant/components/homematicip_cloud/services.py

```diff
@@ -324,6 +344,18 @@ async def _async_reset_energy_counter(hass: HomeAssistant, service: ServiceCall)
                     await device.reset_energy_counter()
 
 
+async def _async_set_home_cooling_mode(hass: HomeAssistant, service: ServiceCall):
+    """Service to set the cooling mode."""
+    cooling = service.data[ATTR_COOLING]
+
+    if hapid := service.data.get(ATTR_ACCESSPOINT_ID):
+        if home := _get_home(hass, hapid):
+            await home.set_cooling(cooling)
+    else:
+        for hap in hass.data[HMIPC_DOMAIN].values():
```

Comment:
Does this mean we would set cooling mode for all "homes" if there is several? Is that a wanted behavior?

Reply:
If an accesspoint ID is given, it only set the cooling mode for that 'home', otherwise for all homes. But you would generally only have 1 home, so the default action doesn't require the access point ID. This is similar to the existing _async_deactivate_vacation